bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/scheduling/Scheduler.java

```diff
@@ -96,6 +96,10 @@ public FutureValue scheduleConsumer(Object[] params, FPValue<?, ?> fp, Strand pa
     public FutureValue schedule(Object[] params, Function function, Strand parent, CallableUnitCallback callback,
                                 Map<String, Object> properties) {
         FutureValue future = createFuture(parent, callback, properties);
+        return schedule(params, function, parent, future);
+    }
+
+    private FutureValue schedule(Object[] params, Function function, Strand parent, FutureValue future) {
```

Comment:
Why do we need this change? Just a refactoring?

Reply:
Yes just a refactoring